sql/updates/world/3.3.5/9999_99_99_99_world_terokk.sql

```diff
@@ -0,0 +1,117 @@
+(-@GUID, 0, 13, 0, 38, 0, 100, 0, 5, 5, 0, 0, 0, 11, 40657, 2, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, "Skyguard Ace - On Data 5 5 Set - Cast 'Ancient Flames'"),
+
+(2337700, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 41, 5000, 0, 0, 0, 0, 0, 10, @GUID+1, 0, 0, 0, 0, 0, 0, 0, "Skyguard Ace - On Script - Despawn In 5 Seconds (Skyguard Ace)"),
+(2337700, 9, 1, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 41, 5000, 0, 0, 0, 0, 0, 10, @GUID+2, 0, 0, 0, 0, 0, 0, 0, "Skyguard Ace - On Script - Despawn In 5 Seconds (Skyguard Ace)"),
+(2337700, 9, 2, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 41, 5000, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, "Skyguard Ace - On Script - Despawn In 5 Seconds");
+
+DELETE FROM `creature_text` WHERE `CreatureID` IN (21838, 23377);
+INSERT INTO `creature_text` (`CreatureID`, `GroupID`, `ID`, `Text`, `Type`, `Language`, `Probability`, `Emote`, `Duration`, `Sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES
+(21838, 0, 0, "Who calls me to this world?  The stars are not yet aligned... my powers fail me!  You will pay for this!", 12, 0, 100, 0, 0, 0, 21639, 0, "Terokk"),
+(21838, 1, 0, "Show me what you're made of, $n!", 14, 0, 100, 0, 0, 0, 21327, 0, "Terokk"),
+(21838, 2, 0, "Kwa! You cannot kill me, I am immortal!", 14, 0, 100, 0, 0, 0, 24020, 0, "Terokk"),
+(21838, 3, 0, "%s becomes enraged as his shield shatters.", 41, 0, 100, 0, 0, 0, 21328, 0, "Terokk"),
+(23377, 0, 0, "Enemy sighted!  Fall into formation and prepare for bombing maneuvers!", 14, 0, 100, 0, 0, 0, 21439, 0, "Skyguard Ace"),
+(23377, 1, 0, "Quickly! Use the flames and support the ground troops. Its ancient magic should cleanse Terokk's shield.", 14, 0, 100, 0, 0, 0, 24021, 0, "Skyguard Ace"),
+(23377, 2, 0, "They did it!  Enemy down!  Return to base!", 14, 0, 100, 0, 0, 0, 21437, 0, "Skyguard Ace");
+
+DELETE FROM `spell_script_names` WHERE `ScriptName`="spell_skyguard_flare";
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
+(40655, "spell_skyguard_flare");
+
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=13 AND `SourceEntry`=40722;
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=22 AND `SourceEntry`=21838;
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES
+(13, 3, 40722, 0, 0, 1, 0, 40726, 0, 0, 1, 0, 0, "", "Will of the Arakkoa God does not affect targets with Chosen One aura"),
+(22, 13, 21838, 0, 0, 1, 1, 40657, 1, 0, 1, 0, 0, "", "SAI executes only if Terokk does not have Ancient Flames aura AND"),
+(22, 13, 21838, 0, 0, 1, 1, 40733, 0, 0, 1, 0, 0, "", "SAI executes only if Terokk does not have Divine Shield aura");
+
+-- Will of the Arakkoa God is negative
+DELETE FROM `spell_custom_attr` WHERE `entry`=40722;
+INSERT INTO `spell_custom_attr` (`entry`, `attributes`) VALUES
+(40722, 4096+8192);
```

Comment:
It's better to fix this in core side, we have lots of similar issues

Reply:
Done! I checked the dbc and the only thing that could tell us it's a negative spell is the target type (TARGET_UNIT_SRC_AREA_ENEMY), maybe that can be useful for a more general fix.

Reply:
@Jildor does this requires the same fix as https://github.com/TrinityCore/TrinityCore/pull/24862/files ?